### PR TITLE
refactor: Queries to Mutations

### LINKED ISSUE #742: 🔧 [Refactor] What is a query? What is a mutation?
## 🔧 Refactor ticket

Refactor the use of query/mutation for our graphql requests. They are distinctively different in behaviour (mutation is blocking, while query is non-blocking). Make sure we use them correctly

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
Makes the queries that alter data in the DB to mutations

### Issues
- fixes #742

Fixes the bug that a user cannot change the language in the profile

### CODE REVIEW

File: frontend/src/views/Pages/Register.spec.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Register from './Register'
 
 const localVue = global.localVue
 
-const resgisterUserQueryMock = jest.fn()
+const resgisterUserMutationMock = jest.fn()
```
Comment (ulfgebhardt) [inline frontend/src/views/Pages/Register.spec.js:8]:
```suggestion
const registerUserMutationMock = jest.fn()
```

File: frontend/src/views/Pages/Register.vue [hunk 2]
```diff
@@ -141,7 +141,7 @@
 import InputEmail from '../../components/Inputs/InputEmail.vue'
 import InputPasswordConfirmation from '../../components/Inputs/InputPasswordConfirmation.vue'
 import LanguageSwitchSelect from '../../components/LanguageSwitchSelect.vue'
-import { resgisterUserQuery } from '../../graphql/queries'
+import { resgisterUser } from '../../graphql/mutations'
```
Comment (ulfgebhardt) [inline frontend/src/views/Pages/Register.vue:144]:
```suggestion
import { registerUser } from '../../graphql/mutations'
```

File: frontend/src/graphql/mutations.js [hunk 3]
```diff
@@ -11,3 +11,59 @@ export const unsubscribeNewsletter = gql`
...
+      password: $password
+      passwordNew: $passwordNew
+      language: $locale
+    ) {
+      validValues
+    }
+  }
+`
+
+export const resgisterUser = gql`
```
Comment (ulfgebhardt) [inline frontend/src/graphql/mutations.js:47]:
```suggestion
export const registerUser = gql`
```

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

**Review by Elweyn — APPROVED** [review]
Seems good for me except the name change (resgister => register) that @ulfgebhardt asked for.

**Comment by Mogge** [comment]
Thx for the catch @ulfgebhardt

**Comment by ulfgebhardt** [comment]
> Thx for the catch @ulfgebhardt

I think that is not a problem introduced in this PR, but I mean its fair to fix it right? ;)

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**